packages/mongodb-rag-ingest/src/commands/all.test.ts

```diff
@@ -142,4 +136,119 @@ describe("allCommand", () => {
       await ingestMetaStore.close();
     }
   });
+
+  it("deletes pages and embedded content that are no longer in the data sources", async () => {
+    const ingestMetaStore = makeIngestMetaStore({
+      connectionUri: uri,
+      databaseName,
+      entryId: "all",
+    });
+    assert(mongoClient);
+
+    // Mock data sources
+    const mockDataSources: DataSource[] = [
+      {
+        name: "source1",
+        fetchPages: async () => [
+          {
+            url: "test1.com",
+            format: "html",
+            sourceName: "source1",
+            body: "hello source 1",
+          },
+        ],
+      },
+      {
+        name: "source2",
+        fetchPages: async () => [
+          {
+            url: "test2.com",
+            format: "html",
+            sourceName: "source2",
+            body: "hello source 1",
+          },
+        ],
+      },
+    ];
+    const mockDataSourceNames = mockDataSources.map(
+      (dataSource) => dataSource.name
+    );
+
+    try {
+      // Start Setup
+      // create pages and verify that they have been created
+      await updatePages({ sources: mockDataSources, pageStore });
+      const pages = await pageStore.loadPages();
+      assert(pages.length == 2);
+      // create embeddings for the pages and verify that they have been created
+      await updateEmbeddedContent({
+        since: new Date(0),
+        embeddedContentStore: embedStore,
+        pageStore,
+        sourceNames: mockDataSourceNames,
+        embedder,
+      });
+      let page1Embedding = await embedStore.loadEmbeddedContent({
+        page: pages[0],
+      });
+      let page2Embedding = await embedStore.loadEmbeddedContent({
+        page: pages[1],
+      });
+      assert(page1Embedding.length);
+      assert(page2Embedding.length);
+      // End Setup
+
+      // run the all command with only one data source and verify that the page and embedding from the other data source are deleted
+      const remainingDataSources = mockDataSources.slice(0, 1);
+      const remainingDataSourcesNames = remainingDataSources.map(
+        (dataSource) => dataSource.name
+      );
+      await doAllCommand(
+        {
+          pageStore: pageStore,
+          embeddedContentStore: embedStore,
+          ingestMetaStore,
+          embedder,
+          dataSources: remainingDataSources,
+        },
+        {
+          async doUpdatePagesCommand() {
+            await doUpdatePagesCommand(
+              {
+                pageStore,
+                dataSources: remainingDataSources,
+                ingestMetaStore,
+                embeddedContentStore: embedStore,
+                embedder,
+              },
+              { source: remainingDataSourcesNames }
+            );
+          },
+        }
+      );
+
+      const [page1, page2] = await pageStore.loadPages();
+      expect(page1).toEqual(
+        expect.objectContaining({
+          url: "test1.com",
+          action: "created",
+        })
+      );
```

Comment:
```suggestion
      expect(page1).toMatchObject(
        {
          url: "test1.com",
          action: "created",
        }
      );
```

nit: more concise and i believe the same effect